Introducing unique states

Boole215 opened the pull request.

Boole215 committed:
Created:
cardGen.js
- "generates" FeedCards using state's cards object

Modified:
App.js
- removed useSelector, useDispatch, FeedCard import
- added CardGen import

card.js
- added props
-- now accesses state info using id passed through props

cardSlice.js
- commented out previous initialState
- added cards object and cardCount int to initialState
- added modified mouse handlers, now accept ID as payload to modify
  individual card states
- added pushID, acts as a constructor for an individual card
- added pushID to action exports

addCard.js
- moved inline onClick logic to reducer

addCardSlice.js
- added addingFeed (boolean) to initialState
- added setters for addingFeed

addDialogue.js
- added styling for component centering and exit button placement

addDialogueSlice.js
- added dialogueField to initialState
- fixed updateField so it actually works with component's state

Boole215 committed:
Updated ReadMe

eddielee394 commented:
it'd probably be best if you added the `.idea` directory to the gitignore file.  No real reason to check that in to version control. 
Since `.idea` has already been committed, you'll have to perform a few additional steps in order to actually ignore it.  Here's a walkthrough:  https://www.git-tower.com/learn/git/faq/ignore-tracked-files-in-git/

Boole215 committed:
created:
mangadexAPI.js
~~~~~~
- currently uses proxy to make api calls
~~~~~~
- client class to keep API logic in one place
- can fetch: Manga, Cover, Chapters, Author

index.css
- added style to remove scrollbar

Modified:

App.js
- added height to feed card container, now 100% height

addDialogue.js
- Modified the TextField's onChange handler, currentValue is no longer stale
- Modified Button onclick, now dispatches an async thunk to fetch series information from the mangadex API

card.js
- modified imageURL, crntTitle, to use info from API

cardSlice.js
- Created async think fetchSeriesByID
-- series of async api requests to populate a feedcard

- Created fulfilled extra reducer fro fetchSeriesByID
-- moves fetched data to state under the series' ID

.gitignore
- added .idea
- added .cert

Boole215 committed:
Merge branch 'dev' of https://github.com/Boole215/mTracker into dev

eddielee394 commented on src/features/addCard/addCard.js:
```diff
@@ -19,11 +19,10 @@ const useStyles = makeStyles({
 
     return(
-        <IconButton size="medium" color="black" onClick={!cantClick && dispatch(toggleAddWind())} className={ classes.addPos }>
+        <IconButton size="medium" color="black" onClick={() => {dispatch(openAddDialogue())}} className={ classes.addPos }>
```
You'll want to avoid calling arrow functions inline like this.  It's not very performant every time the component is rendered a new reference to the arrow function gets generated.  it's also kinda hard to read and navigate about.  Instead something like this would be better:  

```
export function AddCard(){
    const dispatch = useDispatch()
    const classes = useStyles()
    
    const handleOpenDialogue = () => dispatch(openAddDialogue()); 

    return(
        <IconButton size="medium" color="black" onClick={handleOpenDialogue} className={ classes.addPos }>
            <AddIcon fontSize="large" />
        </IconButton>
    )

}
```

Boole215 commented:
Whoops, only meant to approve #3
